### PR TITLE
refactor(mentions): use None instead of _FakeBool

### PR DESCRIPTION
**Pull request by teaishealthy**
## Summary

This PR removes the `_FakeBool` sentinel value used in mentions.py over `None`

## Checklist

- [x] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
  - [x] I have run `task pyright` and fixed the relevant issues.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: nextcord/mentions.py [hunk 1]
```diff
@@ -114,20 +100,20 @@ def to_dict(self) -> AllowedMentionsPayload:
         parse = []
         data = {}
 
-        if self.everyone:
+        if self.everyone or self.everyone is None:
```
Comment (TAG-Epic) [inline nextcord/mentions.py:103]: This seems over complicated. Why not check if its None and make it True?

Comment (teaishealthy) [reply nextcord/mentions.py:103]: It's needed for .merge, which is used throughout the library.

Comment (EmreTech) [reply nextcord/mentions.py:103]: I agree that this is over-complicated, but the proposed solution isn't a good idea. Here's a better approach.
```suggestion
        if self.everyone is not False:
```

Comment (teaishealthy) [reply nextcord/mentions.py:103]: @EmreTech This does not work for checking `self.users`, as that can be List[Snowflake] too

**Review by EmreTech — APPROVED** [review]

**Review by ooliver1 — APPROVED** [review]

**Pull request merged**